src/pubsub/ua_pubsub_writergroup.c

```diff
@@ -188,12 +188,14 @@ UA_Server_freezeWriterGroupConfiguration(UA_Server *server,
         /* PublishedDataSet freezeCounter++ */
         UA_PublishedDataSet *publishedDataSet =
             UA_PublishedDataSet_findPDSbyId(server, dataSetWriter->connectedDataSet);
-        publishedDataSet->configurationFreezeCounter++;
-        publishedDataSet->configurationFrozen = true;
-        /* DataSetFields freeze */
-        UA_DataSetField *dataSetField;
-        TAILQ_FOREACH(dataSetField, &publishedDataSet->fields, listEntry) {
-            dataSetField->configurationFrozen = true;
+        if(publishedDataSet) {
+            publishedDataSet->configurationFreezeCounter++;
+            publishedDataSet->configurationFrozen = true;
+            /* DataSetFields freeze */
+            UA_DataSetField *dataSetField;
+            TAILQ_FOREACH(dataSetField, &publishedDataSet->fields, listEntry) {
+                dataSetField->configurationFrozen = true;
+            }
```

Comment:
Better to pull out pure refactoring out of the functional pull request to comprehend all changes related to the newly introduced feature better.

Reply:
That's not just refactoring. In case of heartbeat, the publishedDataSet pointer would be actually NULL.